lib/chat_review_queue.rb

```diff
@@ -125,6 +136,50 @@ def companion_pm_creator(chat_message, flagger, flag_type_id, opts)
     PostCreator.new(flagger, create_args)
   end
 
+  def find_or_create_transcript(chat_message, flagger, existing_reviewable)
+    previous_message_ids =
```

Comment:
What kind of check do we have to ensure transcript will never surface messages to someone who shouldn't see them?

Reply:
IMO any access access control we add for this should be outside the `ChatTranscriptService`, that should be kept as a dumb thing that just spits out the markdown for the provided IDs. In say ing that it does at least make sure the messages all belong to the same channel so you can't just give it arbitrary IDs. When using the quote in the UI we run an access check on the channel before quoting the messages so that is safe.

In this case, we are sending the context to the moderator group, so it should always be safe here too right?

Reply:
Yes Im not suggesting we add anything to transcript service. Im mostly asking Roman to take a step back and carefuly think about this.

Its potentially super easy to pass down ids to transcript service, so we should double check it and ensures we have tests for this.

Reply:
Yep no problem at all, it's good to have these discussions :+1:

Reply:
The transcript is a PM between the system user and the moderator's group, so all PM restrictions should apply. I'll add some tests to assert they're the only users with access to the PM and prevent possible regressions, but not sure if there's anything else to do here.

Reply:
Im fine with just extra tests :+1: